tests/perf/linux_perf_event.cc

```diff
@@ -81,28 +81,25 @@ linux_perf_event::disable() {
     ::ioctl(_fd, PERF_EVENT_IOC_DISABLE, 0);
 }
 
-linux_perf_event
-linux_perf_event::user_instructions_retired() {
+static linux_perf_event
+make_linux_perf_event(unsigned config, pid_t pid = 0, int cpu = -1, int group_fd = -1, unsigned long flags = 0) {
     return linux_perf_event(perf_event_attr{
             .type = PERF_TYPE_HARDWARE,
             .size = sizeof(struct perf_event_attr),
-            .config = PERF_COUNT_HW_INSTRUCTIONS,
+            .config = config,
             .disabled = 1,
             .exclude_kernel = 1,
             .exclude_hv = 1,
             .exclude_idle = 1,
-            }, 0, -1, -1, 0);
```

Comment:
You're replacing an easy-to-read list of settings with a function that takes a large number of anonymous int parameters with lots of defaults, why?

Reply:
Because it's practically common code.
Other than the `config` field, all other members are the same,
so maintaining 2 open-coded copies require remembering to sync the changes (if needed).

Also, why do you consider specifying `pid_t pid = 0, int cpu = -1, int group_fd = -1, unsigned long flags = 0` in the interface as `anonymous int parameters` while today we pass `0, -1, -1, 0` which is completely cryptic?

Reply:
You're right about pid/cpu/group_fd. I was going to say that the commonality is accidental and the parameters can change independently, but something like disabled/exclude_kernel/exclude_hv isn't so likely to change.